File: forge/src/main/java/dev/biserman/wingscontracts/forge/WingsContractsModForge.java
package dev.biserman.wingscontracts.forge;

import dev.architectury.platform.forge.EventBuses;
import net.minecraftforge.fml.common.Mod;
import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;

import dev.biserman.wingscontracts.WingsContractsMod;

@Mod(WingsContractsMod.MOD_ID)
public final class WingsContractsModForge {
    public WingsContractsModForge() {
        var modBus = FMLJavaModLoadingContext.get().getModEventBus();
        // Submit our event bus to let Architectury API register our content on the right time.
        EventBuses.registerModEventBus(WingsContractsMod.MOD_ID, modBus);

        // Run our common setup.
        WingsContractsMod.init();
    }
}

File: forge/src/main/kotlin/dev/biserman/wingscontracts/forge/WingsContractsModForge.kt
package dev.biserman.wingscontracts.forge

import dev.architectury.platform.forge.EventBuses
import dev.biserman.wingscontracts.WingsContractsMod
import dev.biserman.wingscontracts.compat.CompatMods
import net.minecraft.world.level.block.entity.BlockEntity
import net.minecraftforge.common.MinecraftForge
import net.minecraftforge.fml.ModList
import net.minecraftforge.fml.ModLoadingContext
import net.minecraftforge.fml.common.Mod
import net.minecraftforge.fml.config.ModConfig
import thedarkcolour.kotlinforforge.forge.MOD_BUS
import dev.biserman.wingscontracts.config.ModConfig as WingsContractsConfig

@Mod(WingsContractsMod.MOD_ID)
class WingsContractsModForge {
    init {
        @Suppress("removal")
        ModLoadingContext.get().registerConfig(ModConfig.Type.SERVER, WingsContractsConfig.SERVER_SPEC)

        EventBuses.registerModEventBus(WingsContractsMod.MOD_ID, MOD_BUS)

        WingsContractsMod.init(ForgePlatformHelper())

        if (ModList.get().isLoaded(CompatMods.COMPUTERCRAFT)) {
            ModPeripheralProvider.register()
        }

        MinecraftForge.EVENT_BUS.addGenericListener(
            BlockEntity::class.java,
            PortalItemHandlerCapabilityProvider::attachCapabilities
        )
    }
}
